src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -1132,6 +1132,11 @@ private static int FormatOctal(long value, Span<byte> destination)
         // Writes the specified DateTimeOffset's Unix time seconds, and returns its checksum.
         private int WriteAsTimestamp(DateTimeOffset timestamp, Span<byte> destination)
         {
+            if (timestamp == DateTimeOffset.UnixEpoch)
+            {
+                return 0;
```

Comment:
This seems to write nothing at all and return 0 - is that what you meant by "nulls"?  Should it instead fill with 0 or can you count on the backing span being pre-filled with zeros?

Reply:
Let me check. I'd be surprised if the byte buffer where we write everything is unexpectedly filled with garbage instead of zeroed by default.

Reply:
In the async code we clear the rented buffer to zero it out, because they're not guaranteed to be clean:
https://github.com/dotnet/runtime/blob/518efe04f4f42ff1ec7d3a5a51b2a584ef58fa1d/src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarWriter.cs#L324

Looks like we do the same in the sync code although I don't recall if it's actually necessary (hence the lack of comment): https://github.com/dotnet/runtime/blob/518efe04f4f42ff1ec7d3a5a51b2a584ef58fa1d/src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarWriter.cs#L282

Reply:
In the specific case of the ctime and atime fields, we don't want them to be filled with "0" chars. Instead we want them filled with nulls.

It's the size field the one that we want filled with "0" chars.

Reply:
I believe I answered the concern, but please unresolve the conversation if you have more comments.